ui/src/components/table/table-element/column-settings.tsx

```diff
@@ -0,0 +1,86 @@
+'use client';
+import { Label } from '@/components/ui/label';
+import { Switch } from '@/components/ui/switch';
+import { useState } from 'react';
+import { GroupingPath } from '@/models/groupings-api-results';
+import { Button } from '@/components/ui/button';
+import {
+    DropdownMenu,
+    DropdownMenuTrigger,
+    DropdownMenuContent,
+    DropdownMenuRadioGroup,
+    DropdownMenuRadioItem
+} from '@/components/ui/dropdown-menu';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faSliders } from '@fortawesome/free-solid-svg-icons';
+import { Table } from '@tanstack/table-core';
+
+const ColumnSettings = ({ table }: { table: Table<GroupingPath> }) => {
+    interface ColumnVisibilityState {
+        description: boolean;
+        path: boolean;
+    }
```

Comment:
Can you use `type` here instead of `interface`? Both `type` and `interface` are very similar but `interface` has more benefits when used with classes so it would be better to simply use `type` for this simple declaration.

Reply:
done